polyval/src/lib.rs

```diff
@@ -0,0 +1,205 @@
+//! POLYVAL is a GHASH-like universal hash over GF(2^128) useful for
+//! implementing AES-GCM-SIV (RFC 8452) as well as GMAC/AES-GCM.
+//!
+//! It can be thought of as the little endian equivalent of GHASH, which
+//! affords it a small performance advantage over GHASH when used on little
+//! endian architectures.
+//!
+//! By converting endianness on the way in and out, this means that POLYVAL can
+//! trivially be used to implement GHASH. This means POLYVAL can be used to
+//! implement GMAC, the Message Authentication Code used by AES-GCM.
+
+#![no_std]
+#![doc(html_logo_url = "https://raw.githubusercontent.com/RustCrypto/meta/master/logo_small.png")]
+
+extern crate byteorder;
```

Comment:
Ideally I'd like to get rid of this.

Reply:
If you don't want to wait for the next version of `crypto-mac`, then I think it's fine to keep it for now. You can add a TODO, so we will not forget to remove it on upgrade.

Reply:
I managed to make it optional (gated on the `insecure-soft` cargo feature) and also added a comment about how it can be replaced on 1.32+